base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+set -eu
+
+target_paths=$@
+
+get_linked_libraries() {
+    targets=$@
+
+    executables=$(find $targets -type f -executable -print)
+
+    libs="$(ldd $executables 2>/dev/null | grep '=>' | cut -d'=' -f 1)"
+    for lib in $libs; do
+        echo ${lib%.so*}.so
+    done | sort | uniq
+}
+
+find_shared_libs() {
+    search_paths=$@
+
+    libs=$(find $search_paths -type f -executable -name *.so* -print)
+
+    echo "$libs" | grep -E "*.so(.[0-9]+)*$" | sort | uniq
+}
+
+get_used_epics_modules() {
+    targets=$@
+
+    executables=$(find $targets -type f -executable -print)
+    libs="$(ldd $executables 2>/dev/null)"
+
+    modules=()
+    epics_libs=$(echo "$libs" | grep -oE "$EPICS_MODULES_PATH/.*/*.so")
+    for lib in $epics_libs; do
+        module=${lib%/lib/*/*lib*so}
+        echo $module
+    done | sort | uniq
+}
+
+get_all_epics_modules() {
+    release_defs=$(cat ${EPICS_RELEASE_FILE} | grep = | cut -d'=' -f 2)
+
+    modules="$(echo "$release_defs" | grep $EPICS_MODULES_PATH)"
+
+    for module in $modules; do
+        echo $module
+    done
+}
+
+remove_static_libs() {
+    for target; do
+        libs=$(find $target -type f -name *.a -print)
```

Comment:
I'm assuming there are only [AR archives](https://en.wikipedia.org/wiki/Ar_(Unix)) which are statically linked libraries. Is that a really good assumption?

Reply:
For running on Linux, I think it should be.

If you want to have a sort of lint for our build process, after the base image is done building stuff, I'd run a script to make sure we don't have any `.o` files leftover. They are the only other kind of "static" files we could have.

Reply:
I've checked our base image, and it does not have any `*.o` left behind. Our cleaning done after installation seems to be enough for now to remove these.